doschema/jsvalidation.py

```diff
@@ -0,0 +1,310 @@
+
+        :param field_value: Type of the field.
+        :param curr_field: Tuple with path to currently processed field.
+        """
+        field_path = curr_field
+        if isinstance(field_value, list):
+            for elem in field_value:
+                if self.ignore_index:
+                    new_field_path = field_path
+                else:
+                    new_field_path = field_path + (field_value.index(elem), )
+
+                self._validate_root(elem, new_field_path)
+
+        elif isinstance(field_value, dict):
+            self._validate_root(field_value, curr_field)
+
+    def _validate_collection(self, curr_schema, curr_field):
+        """Check schema for allOf, anyOf and oneOf.
+
+        :param curr_schema: Schema or subschema that is currently processed.
+        :param curr_field: Tuple with path to currently processed field.
+        """
+        keys = self.COLLECTION_KEYS.intersection(curr_schema)
+        for key in keys:
+            subschema = curr_schema[key]
+
+            for elem in subschema:
+                self._validate_root(elem, curr_field)
+
+    def _validate_dependencies(self, curr_schema, curr_field):
+        """validate_root and check dependencies in schema.
+
+        :param curr_schema: Schema or subschema that is currently processed.
+        :param curr_field: Tuple with path to currently processed field.
+        """
+        dependencies = curr_schema['dependencies']
+
+        while '$ref' in dependencies:
+            path = dependencies.get('$ref')
+            dependencies = self.resolver.resolve(path)[1]
+
+        for field_name, field_value in six.iteritems(dependencies):
+            if curr_field + (field_name, ) not in self.fields_types_dict:
+                self.fields_types_dict[
+                    curr_field + (field_name, )
+                ] = FieldToAdd(
+                        schema_index=self.uri,
+                        field_tuple=curr_field + (field_name, ),
+                        field_type=None
+                    )
+
+            if isinstance(field_value, dict):
+                self._validate_root(field_value, curr_field)
+
+            elif type(field_value) is list:
+                for field in field_value:
+                    path = curr_field + (field, )
+                    if path not in self.fields_types_dict:
+                        self.fields_types_dict[
+                            curr_field + (field_name, )
+                        ] = FieldToAdd(
+                                schema_index=self.uri,
+                                field_tuple=curr_field + (field_name, ),
+                                field_type=None
+                            )
+            else:
+                raise ValueError(
+                    'Dependencies value is not a dict nor a list.'
+                )
+
+    def _validate_enum(self, field_value, path):
+        """Process each element in enum field according to ignore_index.
```

Comment:
Please replace this docstring with:
"""Process each element in enum field values."""

`ignore_index` is only used for array type.